src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1322,46 +1322,24 @@ int fmi3_xml_handle_FloatXXVariable(fmi3_xml_parser_context_t* context, const ch
         if (!declaredType) return -1;
 
         /* Set type properties */
-        {
-            int hasUnit = fmi3_xml_is_attr_defined(context, fmi_attr_id_unit)
-                       || fmi3_xml_is_attr_defined(context, fmi_attr_id_displayUnit);
-            int hasMin  = fmi3_xml_is_attr_defined(context, fmi_attr_id_min);
-            int hasMax  = fmi3_xml_is_attr_defined(context, fmi_attr_id_max);
-            int hasNom  = fmi3_xml_is_attr_defined(context, fmi_attr_id_nominal);
-            int hasQuan = fmi3_xml_is_attr_defined(context, fmi_attr_id_quantity);
-            int hasRelQ = fmi3_xml_is_attr_defined(context, fmi_attr_id_relativeQuantity);
-            int hasUnb  = fmi3_xml_is_attr_defined(context, fmi_attr_id_unbounded);
-
-            if (hasUnit || hasMin || hasMax || hasNom || hasQuan || hasRelQ || hasUnb) {
-                /* create a new type_props that overrides declared type's properties when necessary */
-
-                fmi3_xml_float_type_props_t* dtProps = 0; /* declaredType properties */
-
-                if (declaredType->structKind == fmi3_xml_type_struct_enu_typedef)
-                    dtProps = (fmi3_xml_float_type_props_t*)(declaredType->nextLayer);
-                else /* default type */
-                    dtProps = (fmi3_xml_float_type_props_t*)declaredType;
-
-                fmi3_xml_reserve_parse_buffer(context, 1, 0);
-                fmi3_xml_reserve_parse_buffer(context, 2, 0);
-
-                type = fmi3_xml_parse_float_type_properties(context, elmID, defaultType, primType);
-                if (!type) {
-                    variable->type = &((fmi3_xml_float_type_props_t*)declaredType)->super; /* fallback to default */
-                    return -1;
-                }
-
-                type->super.nextLayer = declaredType;
-                if (!hasUnit) type->displayUnit              = dtProps->displayUnit;
-                if (!hasMin)  type->typeMin                  = dtProps->typeMin;
-                if (!hasMax)  type->typeMax                  = dtProps->typeMax;
-                if (!hasNom)  type->typeNominal              = dtProps->typeNominal;
-                if (!hasQuan) type->quantity                 = dtProps->quantity;
-                if (!hasRelQ) type->super.isRelativeQuantity = dtProps->super.isRelativeQuantity;
-                if (!hasUnb)  type->super.isUnbounded        = dtProps->super.isUnbounded;
```

Comment:
All of this is handled in the the parse_float_type_properties. Before it used the default type for non-set optional attrs. Now it immediately uses the declaredType.